mm0-rs/README.md

```diff
@@ -6,19 +6,80 @@ Watch this space for more updates, as `mm0-rs` is under active development. See
+You then should make it easy to execute.
+One way is to copy or symlink the resulting executable file `target/release/mm0-rs` to your system path.
+An alternative, if you're integrating with Visual Studio Code as suggested
+below, is to point `vscode-mm0` to the executable using (as
+explained below) the setting "metamath-zero.executablePath": "mm0-rs"
+in your `settings.json` file.
 
-    "metamath-zero.executablePath": "mm0-rs",
+## Integration with Visual Studio Code
 
-in your `settings.json` file.
+We typically use Visual Studio Code, an open source software
+editor / IDE which supports LSP servers.
+If you choose to do the same,
+[download and install Visual Studio Code](https://code.visualstudio.com/).
+
+Select View/Extensions, search for "Metamath"
+Look for "Metamath Zero" & press its "install" button.
+
+If `mm0-rs` is not on your path, then you need to
+change the Visual Studio Code settings so it can execute `mm0-rs`.
+To do this, after you've installed "Metamath Zero" look at the name,
+click on its small "gear symbol" to change its settings.
+Note: if you click on its "gear symbol" and see nothing, that's a bug
+in the Visual Studio Code program; restart it and again select View/Extensions
+to you can see the settings to change them.
+Now modify the "Metamath-zero Executable Path" setting (which corresponds
+to `metamath-zero.executablePath` in `settings.json`)
+to be the correct path for the executable
+(it defaults to `mm0-rs`).
+You should probably use an absolute path (e.g., one starting with "/"
+on Linux/Unix/MacOS).
 
 ## Usage
 
+The `mm0-rs` program provides a number of modes and options if you
+want to invoke it directly. For example:
+
 * `mm0-rs server` causes it to send and receive LSP server commands via stdin and stdout. This is not used directly from the CLI but rather is invoked by `vscode-mm0` when it is set up to use `mm0-rs` as a language server.
-  * `mm0-rs server --debug` is run by `vscode-mm0` when the extension itself is run in debugging mode, and this will enable backtraces and logging.
+* `mm0-rs server --debug` is run by `vscode-mm0` when the extension itself is run in debugging mode, and this will enable backtraces and logging.
+* `mm0-rs compile foo.mm1` will compile an MM1 file, reporting errors to the console. This is essentially the console version of the `server` mode.
+
+You can easily use `mm0-rs` from within Visual Studio Code.
+Start Visual Studio Code, then use File/Open,
+and navigate to the mm0/examples directory.
+Select a file to look at; `do-block.mm1` and `demo.mm1`
+are good starting points.
+Select "View/Problems" to see a summary.
+
+File `demo.mm1` show what proofs look like in MM1.
+A theorem proof looks like `theorem NAME: $ ... expression... $ =`
+followed by an s-expression-like proof.
+"(stat)" prints the current proof state.
+
+There aren't too many fancy tactics right now unless you write them
```

Comment:
This paragraph and the next one don't really make sense outside of the context of the email I sent you. I think they should just be cut, and you should see how to modify mm1.md to include this information if not already present, or a tutorial extracted if that's what you are going for.

Reply:
Moving them into mm1.md is fine, though that file should probably be in "docs" instead of "mm0-hs".

However, I think it's important that this information be somewhere.

I'd love to write a tutorial, but I don't understand your system well enough yet to write one :-).

I'll submit an updated PR to try to do this.

Reply:
demo.mm1 was initially designed as a tutorial where I spoke while showing the code. It can probably be tweaked to add more text and make it a proper tutorial.